packages/app/src/systems/Transaction/services/transaction.tsx

```diff
@@ -318,29 +315,58 @@ export class TxService {
   static async getTransactionHistory({
     address,
     providerUrl = '',
+    pagination,
   }: TxInputs['getTransactionHistory']) {
-    const provider = await createProvider(providerUrl || '');
-
+    const provider = await createProvider(providerUrl);
     const txSummaries = await getTransactionsSummaries({
       provider,
       filters: {
         owner: address,
-        first: 50,
+        first: TXS_PER_PAGE,
+        after: pagination?.after,
       },
     });
 
-    const sortedTransactions = txSummaries.transactions?.sort((a, b) => {
-      const aTime = bn(a.time, 10);
-      const bTime = bn(b.time, 10);
-      return aTime.gt(bTime) ? -1 : 1;
-    });
-
     return {
-      transactionHistory: sortedTransactions,
+      transactionHistory: txSummaries.transactions,
       pageInfo: txSummaries.pageInfo,
     };
   }
 
+  static async getAllCursors({
+    address,
+    providerUrl = '',
+    initialEndCursor,
+  }: TxInputs['getAllCursors']) {
+    const provider = await createProvider(providerUrl);
+
+    let hasNextPage = true;
+    const cursors: string[] = [];
+    let endCursor: string | null | undefined = initialEndCursor;
+
+    while (hasNextPage) {
+      const { pageInfo } = await getTransactionsSummaries({
+        provider,
+        filters: {
+          owner: address,
+          first: TXS_PER_PAGE,
+          after: endCursor,
+        },
+      });
+
+      hasNextPage = pageInfo.hasNextPage;
+      endCursor = pageInfo.endCursor;
+
+      if (hasNextPage && endCursor) {
+        cursors.push(endCursor);
+      }
+    }
+
+    return {
+      cursors,
+    };
+  }
+
```

Comment:
Shouldn't we limit this cache to only 200 txs?

Reply:
actually I don't think it's really possible to limit exactly in `200` due to the cursor pagination.
since to get the latest page, we will need to get also the first. it means iterating all of them anyway.

i think we're fine keeping this (bridge is doing the same also)

Reply:
no @arthurgeron the final decision will be to get all